lib/kernel/src/application_controller.erl

```diff
@@ -344,7 +344,7 @@ get_pid_env(Master, Key) ->
 
 get_env(AppName, Key) ->
```

Comment:
This is used in two ways from `application`. In particular `application:get_env/3` has a default value, which could use `ets:lookup_element` directly with no wrapping.

It might make sense to export 2 functions from `application_controller` - one that returns the `{ok, _} | undefined` pattern, one that returns `_ | Default`. This would have the additional benefit of avoiding allocating the `{ok, _}` tuple on lookup successes.

Reply:
@RobinMorisset I suggest that you implement this suggestion and that you use a reference in the version that has to return an `ok` tuple.

Reply:
Done, thanks for the suggestions and comments.